### ISSUE TITLE
Filter by arbitrary parameter

### ISSUE THREAD

**Issue by nicodmf** [api-platform/admin#322]
Hello,

I have a custom filter defined on our API, the parameter name is modificationRestrict and not rely on a property. I see that the code hide this filter because the schemaAnalyser doesn't find it in the property list. I don't understand why this choice has been made, but it's seems this restrict to much the admin.

As I don't know if this is a bug or a design choice and consequently a feature request, I send a support question.

**Comment by alanpoulain** [comment]
It's a design choice: the admin needs to receive the filter information from the API in order to display it correctly.

**Comment by nicodmf** [comment]
As it, the filter receive from api are presents, the admin is aware from there. You can see them in the ["hydra:search"]["hydra:mapping"] path. And the filter exists too on the rest api. So it seems to me that the admin receive enough informations about theres.

![image](https://user-images.githubusercontent.com/173332/91813297-ce301900-ec32-11ea-8959-5ad9fd782e39.png)

**Comment by alanpoulain** [comment]
You're talking about this line then: https://github.com/api-platform/admin/blob/7a266802656e04f4c7ef1ccba17f7e445b260bd0/src/hydra/schemaAnalyzer.js#L58
I think it has been done like this to prevent showing filters if a user doesn't have the rights to display some fields.

**Comment by nicodmf** [comment]
Yes i think this is this line, but authorized field is just the fields mention by this line : 

https://github.com/api-platform/admin/blob/7a266802656e04f4c7ef1ccba17f7e445b260bd0/src/hydra/schemaAnalyzer.js#L50

Do you see a method that permit to see the filters ? (without add a empty property on the model).

**Comment by nicodmf** [comment]
@alanpoulain ping :)

**Comment by DerDu** [comment]
To further bump up the importance of this topic i give another example

```
#[QueryParameter(key: 'filter-:property', filter: new PartialSearchFilter(), properties: ['description'])]
class Test {
    private ?string $description = null;
    ...
}
```
This is a valid (and needed) api-platform declaration
This will work fine using `/tests?filter-description=example` but will not show up in api-admin

In general, if the 'key' doesn't equal the class property - it doesn't work
Despite the information about the mapping is available
```
   "search": {
        "@type": "IriTemplate",
        "template": "/tests{?filter-description}",
        "variableRepresentation": "BasicRepresentation",
        "mapping": [
            {
                "@type": "IriTemplateMapping",
                "variable": "filter-description",
                "property": "description",
                "required": false
            },
        ]
    }
```
